spec/models/polling/cert_status_updater_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'rails_helper'
+require 'polling/cert_status_updater'
+
+RSpec.describe CertStatusUpdater, type: :model do
+
+  include CertStatusUpdater
+
+  let(:root) { PKI.new }
+
+  let(:msa_enc_cert_1) { create(:msa_encryption_certificate, in_use_at: nil) }
+  let(:msa_enc_cert_2) { create(:msa_encryption_certificate, in_use_at: nil) }
+  let(:msa_sgn_cert_1) { create(:msa_signing_certificate, in_use_at: nil) }
+  let(:msa_sgn_cert_2) { create(:msa_signing_certificate, in_use_at: nil) }
+  let(:msa_sgn_cert_3) { create(:msa_signing_certificate, in_use_at: nil) }
+
+  let(:service_1) { create(:service, entity_id: 'entity 1') }
+  let(:service_2) { create(:service, entity_id: 'entity 2') }
+  let(:service_3) { create(:service, entity_id: 'entity 3') }
+  let(:sp_component_1) { create(:sp_component, services: [service_1]) }
+  let(:sp_component_123) { create(:sp_component, services: [service_1, service_2, service_3]) }
+  let(:sp_enc_cert_for_single_entity) { create(:sp_encryption_certificate, component: sp_component_1, in_use_at: nil) }
+  let(:sp_enc_cert_for_multiple_entities) { create(:sp_encryption_certificate, component: sp_component_123, in_use_at: nil) }
+  let(:sp_sgn_cert_for_single_entity) { create(:sp_signing_certificate, component: sp_component_1, in_use_at: nil) }
+  let(:sp_sgn_cert_for_multiple_entities) { create(:sp_signing_certificate, component: sp_component_123, in_use_at: nil) }
+  let(:sp_sgn_cert_other) { create(:sp_signing_certificate, in_use_at: nil) }
+
+  describe '#update_hub_usage_status_for_cert' do
+    context 'When given an encryption cert belonging to an MSA component' do
+      it "leaves the cert unchanged if a call to the Hub does not show it is in use" do
+        allow_any_instance_of(HubConfigApi).to receive(:encryption_certificate).and_return(msa_enc_cert_2.x509)
```

Comment:
I'm following a colleague's suggestion to do it this way, but the internet seems to think that `allow_any_instance_of` is terrible (and I can see their point).  If this were Java, I'd be injecting an instance of HubConfigApi into my module and stubbing that directly; I'm not sure what the idiomatic Ruby equivalent is though.

Reply:
We're now injecting a HubConfigApi instance, which means the code can be tested without using `allow_any_instance_of`.